### PR TITLE
completed view-entries feature of code journal.

### LINKED ISSUE #2: User can view their entries.
## 💰 Motivation

Once a user records some journal entries, they'll want to see them.

## 📐 Plan

Display the user's journal entries in reverse chronological order so they can see the newest ones first.

## 🖼️ Wireframes

user can view their entries mobile
user can view their entries desktop

## ✅ Task List

- [ ] Add an empty unordered list to the `div[data-view="entries"]` element in the HTML.
- [ ] Create some dummy journal entries to the unordered list in the HTML and style them to match the example entries in the figma. Each entry’s content should be in a single `li` element within the `ul` element.
- [ ] Check that the example entries you just created in the unordered list are responsive for both mobile and desktop screens.
- [ ] Define a function named `renderEntry` with a single parameter named `entry`. This parameter represents a single object for a single entry from the `data.entries` array. The function should generate and return a DOM tree for a single entry that matches the entries you created in the unordered list.
- [ ] Create an event listener on the `document` which listens for the `DOMContentLoaded` event. In the callback function for that event, loop through the `data.entries` array and generate a DOM tree for each entry in the array, and append that DOM tree to the unordered list.
- [ ] Remove the example HTML entries from `index.html` once the user's entries are successfully being appended to the DOM on page load and replace them with a message indicating that no entries have been recorded which matches the figma.
- [ ] Create a function named `toggleNoEntries` which toggles the no entries text to show or hide when the function is called.
- [ ] Create a new function named `viewSwap` with a single parameter representing the name of the view to show (the value will be either `”entries”` or `”entry-form”`). This function should show the view whose name was provided as an argument, as well as assign the string argument to the `data.view` property so that the currently shown view is tracked in the data model for the application.
- [ ] Add an anchor to the navbar to show the `”entries”` view. You will need to create an event handler function that utilizes the `viewSwap` function to show the proper view.
- [ ] Add an anchor to the entries view to show the `entry-form`. It should be styled like a button with the text `NEW`, and must match the figma.
- [ ] Update the `submit` callback for the form to do the following:
  - [ ] Render a DOM tree for the newly submitted entry object using the `renderEntry` function.
  - [ ] Prepends the new DOM tree to the unordered list.
  - [ ] Use the `viewSwap` to show the `”entries”` view.
  - [ ] conditionally uses the `toggleNoEntries` function as needed to remove the no entries text.
- [ ] Update the `DOMContentLoaded` callback function to do the following:
  - [ ] uses the `viewSwap` function to show the view which was displayed prior to page refresh.
  - [ ] conditionally uses the `toggleEntries` function as needed to show or remove the no entries text.
- [ ] Clean up all unnecessary styling.
- [ ] Test the app some more in a private browsing tab.
- [ ] Fix all errors.

### PR DESCRIPTION
**Pull request by Katelyn-Ngo**
Recording 2024-06-05 at 16 59 01

### CODE REVIEW

**Comment by Katelyn-Ngo** [comment]
Recording 2024-06-05 at 17 02 55

File: ts/main.ts [hunk 1]
```diff
@@ -1,14 +1,23 @@
 document.addEventListener('DOMContentLoaded', () => {
   const $form = document.getElementById('entry-form') as HTMLFormElement;
   const $titleInput = document.getElementById('title') as HTMLInputElement;
-
   const $photoUrlInput = document.getElementById(
     'photo-url',
```
Comment (brandon-moy) [inline ts/main.ts]: I may have missed this earlier, but your `DOMContentLoaded` event listener should not be over all of your `ts` file. Your `DOMContentLoaded` event listener should just have the part where you are appending entries to the DOM, swapping your view, and then toggling no entries.

File: ts/main.ts [hunk 2]
```diff
@@ -28,9 +37,93 @@ document.addEventListener('DOMContentLoaded', () => {
...
+    event.preventDefault();
+    viewSwap('entry-form');
+  });
+
+  if ($entriesLink) {
+    $entriesLink.addEventListener('click', (event) => {
+      event.preventDefault();
+      viewSwap('entries');
+    });
+  }
```
Comment (brandon-moy) [inline ts/main.ts]: Why do you have these in two different ways? One you are not validating to add the event listener but the other you are only adding the event listener if it have a value? Instead you should do what we did before and throw an error if the value is falsy so that you can know if/where this goes wrong.
```
if (!$entriesLink) throw new Error("error querying $entriesLink")

$entriesLink.addEventListener. . .
```

File: ts/main.ts [hunk 3]
```diff
@@ -28,9 +37,93 @@ document.addEventListener('DOMContentLoaded', () => {
...
+  $entryItem.appendChild($entryContent);
+
+  return $entryItem;
+}
+
+function toggleNoEntries(): void {
+  const $entriesList = document.getElementById(
+    'entries-list',
+  ) as HTMLUListElement;
+  const $noEntries = document.getElementById('no-entries') as HTMLElement;
```
Comment (brandon-moy) [inline ts/main.ts]: Why not query for these outside of the function? If these are elements in your HTML that are always present, we don't need to query for them in a function because we know they will be there. If we have it in the function we are adding more work to have to keep requerying the DOM for something that is permanent.

tldr; if the elements are permanently present on the HTML, move these to the top of the file

File: ts/main.ts [hunk 4]
```diff
@@ -28,9 +37,93 @@ document.addEventListener('DOMContentLoaded', () => {
...
+  }
+}
+
+function viewSwap(view: string): void {
+  const $entryFormView = document.querySelector(
+    '[data-view="entry-form"]',
+  ) as HTMLElement;
+  const $entriesView = document.querySelector(
+    '[data-view="entries"]',
+  ) as HTMLElement;
```
Comment (brandon-moy) [inline ts/main.ts]: Same thing with these, these are available on the page when the HTML loads so these queries can be moved to the top of the file

**Review by brandon-moy — CHANGES_REQUESTED** [review]
Logically in your TypeScript you did really well! Most of the changes I want are not fully related to logic but do help to optimize your code with error handling and efficiency.

For the styling though here are a few things:
- I think you have your mobile and desktop switch, on desktop the text should be on the side of the image on Entries page and on mobile it should be below
- You images are too big. You did not give a width to the images so they are taking up the width of the image instead of taking up the width of the container. I think also adding width to your `.form-group` will help with the image on the create form

**Comment by Katelyn-Ngo** [comment]
Recording 2024-06-06 at 09 48 06

**Review by brandon-moy — APPROVED** [review]
Great job, approved! 🔥
